### PR TITLE
Rose bunch tutorial

### PR DESCRIPTION
**Pull request by arjclark**
Adds in a tutorial for the new rose bunch built in app.

@kaday please review 1
@matthewrmshin please review 2

### CODE REVIEW

File: doc/rose-rug-advanced-tutorials-bunch.html [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+      into the bin directory of the <samp>lander</samp> app and running:</p>
+      <pre class="shell">
+chmod +x land
+</pre>
+    </div>
+
+    <div class="slide">
+      <h3 class="alwayshidden">Run Example</h3>
+
+      <p>Navigate to the root directory of your suite and run <samp>rose
```
Comment (kaday) [inline doc/rose-rug-advanced-tutorials-bunch.html]: You may wish to qualify this as being the level of the suite with the suite.rc and rose-suite.conf.

File: doc/rose-rug-advanced-tutorials-bunch.html [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+      <p>This configuration will run a <code>rose bunch</code> task that calls
+      multiple instances of the <code>land</code> command, supplying arguments
+      to each instance from the <var>class</var> and <var>passengers</var>
+      entries under <var>[bunch-args]</var>.</p>
+    </div>
+
+    <div class="slide">
+      <h3 class="alwayshidden">Example App (3)</h3>
+
+      <p>In the <var>app/lander/</var> directory create a <samp>bin/</samp>
```
Comment (kaday) [inline doc/rose-rug-advanced-tutorials-bunch.html]: Your level of instruction is inconsistent, I would prefer it if you followed the same level of instructions all the way through, if you are going to tell them to `cd` and `mkdir` then you should do it throughout the tutorial.

File: doc/rose-rug-advanced-tutorials-bunch.html [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+    </div>
+
+    <div class="slide">
+      <h3 class="alwayshidden">Run Example</h3>
+
+      <p>Navigate to the root directory of your suite and run <samp>rose
+      suite-run</samp>. Your suite should run, launch the cylc gui and
+      successfully run the lander app.</p>
+
+      <p>Once the suite has finished running and has shutdown, open <samp>rose
```
Comment (kaday) [inline doc/rose-rug-advanced-tutorials-bunch.html]: Add that they can close the gcylc GUI?

File: doc/rose-rug-advanced-tutorials-bunch.html [hunk 4]
```diff
@@ -0,0 +1,333 @@
...
+      bush</samp> to view its output. You can quickly get to the relevent page
+      by running <samp>rose slv</samp> in the suite directory.</p>
```
Comment (kaday) [inline doc/rose-rug-advanced-tutorials-bunch.html]: `suite directory` here, `root directory` above, I would prefer that it was consistent, or if you must `root directory, also referred to as the suite directory where the suite.rc and the rose-suite.conf are located`

File: doc/rose-rug-advanced-tutorials-bunch.html [hunk 5]
```diff
@@ -0,0 +1,333 @@
...
+    <div class="slide">
+      <h3 class="alwayshidden">Limiting concurrent invocations (2)</h3>
+
+      <p>In the case of our simulated airport we will pretend we only have two
+      runways available at a time on which our planes can land. As such we need
+      to limit the number of planes being landed. We do this using the
+      <code>[bunch]pool-size=</code> configuration option of the <code>rose
+      bunch</code> app.</p>
+
+      <p>Open up your app config and add the following line to the
```
Comment (kaday) [inline doc/rose-rug-advanced-tutorials-bunch.html]: What `app config`?

File: doc/rose-rug-advanced-tutorials-bunch.html [hunk 6]
```diff
@@ -0,0 +1,333 @@
...
+pool-size=2
+</pre>
+    </div>
+
+    <div class="slide">
+      <h3 class="alwayshidden">Limiting concurrent invocations (3)</h3>
+
+      <p>Run your suite again. Notice that this time round it takes longer for
+      the task to run as it has been limited in the number of tasks it can run
+      at once. You can see the individual commands being started by viewing the
```
Comment (kaday) [inline doc/rose-rug-advanced-tutorials-bunch.html]: `at once` to `at any one time` or `at the same time` or `simultaneously` etc.

File: doc/rose-rug-advanced-tutorials-bunch.html [hunk 7]
```diff
@@ -0,0 +1,333 @@
...
+      task stdout in the cylc gui. For example, when the "BA007" invocation
```
Comment (kaday) [inline doc/rose-rug-advanced-tutorials-bunch.html]: How do they see the stdout in the gui? Explain.

File: doc/rose-rug-advanced-tutorials-bunch.html [hunk 8]
```diff
@@ -0,0 +1,333 @@
...
+      learnt how to name the individual variants for convenience in examining
+      the logs and how to limit the number of concurrently running
+      commands.</p>
+
+      <p>Further options are listed in the <code>rose bunch</code> <a href=
+      "rose-rug-task-run.html#rose-task-run.util.rose_bunch">documentation</a>.
+      These include, configuring how to proceed following failure of an
+      individual command invocation - <code>[bunch]fail-mode=</code>),
+      automatically generating <var>N</var> <code>command-instances</code> and
+      enabling/disabling the app's incremental mode.</p>
```
Comment (kaday) [inline doc/rose-rug-advanced-tutorials-bunch.html]: `mode etc.`

**Comment by kaday** [comment]
Works okay, just needs some tweaking.

File: doc/rose-rug-advanced-tutorials-bunch.html (hunk 2 above)
Comment (arjclark) [inline doc/rose-rug-advanced-tutorials-bunch.html]: Done.

**Comment by arjclark** [comment]
@kaday - I think I've covered everything.

**Comment by kaday** [comment]
Looks okay to me with the changes that have been made.
Over to @matthewrmshin

File: doc/rose-rug-advanced-tutorials-bunch.html [hunk 9]
```diff
@@ -0,0 +1,341 @@
...
+    UTC mode = True # Ignore DST
+    [[event hooks]]
+        timeout handler = rose suite-hook --mail --shutdown
+        timeout = P3D
+[scheduling]
+    [[dependencies]]
+        graph = lander
+[runtime]
+    [[root]]
+        command scripting = rose task-run
```
Comment (matthewrmshin) [inline doc/rose-rug-advanced-tutorials-bunch.html]: Please update this to `script`, to be consistent with change in #1765.

File: doc/rose-rug-advanced-tutorials-bunch.html [hunk 10]
```diff
@@ -0,0 +1,341 @@
...
+    </div>
+
+    <div class="slide">
+      <h3 class="alwayshidden">Example suite.rc</h3>
+      <pre class="prettyprint lang-cylc">
+[cylc]
+    UTC mode = True # Ignore DST
+    [[event hooks]]
+        timeout handler = rose suite-hook --mail --shutdown
+        timeout = P3D
```
Comment (matthewrmshin) [inline doc/rose-rug-advanced-tutorials-bunch.html]: You may want to remove these event hooks. They don't add value to the tutorial, and we can now rely on individual sites to have these configured to suit their needs.

**Comment by arjclark** [comment]
@matthewrmshin - done and done.

**Comment by matthewrmshin** [comment]
I knew I should have run the test battery - `t/syntax/01-doc.t` is now failing. Please fix with a new pull request.

**Comment by arjclark** [comment]
Ah, my apologies must've not had the new test in my branch.

**Pull request merged**